example_app_frames/src/main/java/com/checkout/example/frames/ui/screen/ThreedSecureActivity.kt

```diff
@@ -0,0 +1,81 @@
+package com.checkout.example.frames.ui.screen
+
+import android.content.Context
+import android.os.Bundle
+import android.view.ViewGroup
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.ui.platform.LocalContext
+import com.checkout.CheckoutApiServiceFactory
+import com.checkout.base.model.Environment
+import com.checkout.example.frames.ui.utils.FAILURE_URL
+import com.checkout.example.frames.ui.utils.PUBLIC_KEY
+import com.checkout.example.frames.ui.utils.PromptUtils
+import com.checkout.example.frames.ui.utils.PromptUtils.neutralButton
+import com.checkout.example.frames.ui.utils.SUCCESS_URL
+import com.checkout.example.frames.ui.utils.URL_IDENTIFIER
+import com.checkout.frames.R
+import com.checkout.threedsecure.model.ThreeDSRequest
+import com.checkout.threedsecure.model.ThreeDSResult
+import com.checkout.threedsecure.model.ThreeDSResultHandler
+
+class ThreedSecureActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            val url = intent.getStringExtra(URL_IDENTIFIER)
+            invokeThreedSecureFlow(LocalContext.current, url ?: "", findViewById(android.R.id.content))
+        }
+    }
+
+    private fun showAlertDialog(componentActivity: ComponentActivity, title: String, message: String) {
+        PromptUtils.alertDialog(componentActivity) {
+            setTitle(title)
+            setMessage(message)
+            neutralButton {
+                it.dismiss()
+                componentActivity.finish()
+            }
+        }.show()
+    }
+
+    private fun invokeThreedSecureFlow(context: Context, url: String, viewGroup: ViewGroup) {
+        /**
+         * Creating instance of CheckoutApiClient
+         */
+        val checkoutApiClient = CheckoutApiServiceFactory.create(
```

Comment:
Is it okay that we use headless SDK directly?
Why not to use frames instead?

Reply:
We can use but we need to add an unnecessary set to the mPaymentFormMediator and configuration for it. At the same time, we will provide merchant to set an example of how we can use headless SDK for 3DS support.